mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -707,16 +707,23 @@ def _log_metrics(self, run_id, metrics):
         for metric in metrics:
             metric, value, is_nan = self._get_metric_value_details(metric)
             if metric not in seen:
-                metric_instances.append(
-                    SqlMetric(
-                        run_uuid=run_id,
-                        key=metric.key,
-                        value=value,
-                        timestamp=metric.timestamp,
-                        step=metric.step,
-                        is_nan=is_nan,
+                try:
+                    metric_instances.append(
+                        SqlMetric(
+                            run_uuid=run_id,
+                            key=metric.key,
+                            value=float(value),
+                            timestamp=metric.timestamp,
+                            step=metric.step,
+                            is_nan=is_nan,
+                        )
                     )
-                )
+                except ValueError as exc:
```

Comment:
Instead of wrapping the entire statement here in a try catch, can you define a new variable that attempts the casting and catches the ValueError from that casting only? 
Right now there are 3 separate actions that are being wrapped (and any one of them could potentially throw a ValueError that isn't related to what you're trying to catch): The cast to float (which you want to catch), the initialization of a SqlMetric instance, and an append command to the `metric_instances` collection.